v2/installer.ps1

```diff
@@ -0,0 +1,574 @@
+
+$dataSrc = @()
+
+foreach ($mc in (Get-AppxPackage -Name "Microsoft.Minecraft*")) {
+    $dataSrc += [PSCustomObject]@{
+        FriendlyName    = (Get-AppxPackageManifest -Package $mc).Package.Properties.DisplayName
+        InstallLocation = $mc.InstallLocation
+    }
+}
+
+function Backup-InitialShaderFiles() {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]$Location,
+        [Parameter(Mandatory = $false)]
+        [string[]]$Materials = @(
+            "RTXStub.material.bin",
+            "RTXPostFX.Tonemapping.material.bin"
+        ),
+        [Parameter(Mandatory = $false)]
+        [string]$BackupDir = "$BRTX_DIR\backup"
+    )
+
+    $mcSrc = "$Location\data\renderer\materials"
+
+    if (-not (Test-Path $BackupDir)) {
+        New-Item -ItemType Directory -Path $BackupDir -Force | Out-Null
+    }
+
+    foreach ($file in $Materials) {
+        $src = "$mcSrc\$file"
+        $dest = "$BackupDir\$file"
+
+        if (Test-Path $src) {
+            Copy-Item -Path $src -Destination $dest -Force -ErrorAction Stop
+        }
+    }
+}
+
+function Backup-ShaderFiles() {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]$Location,
+        [Parameter(Mandatory = $false)]
+        [string[]]$Materials = @(
+            "RTXStub.material.bin",
+            "RTXPostFX.Tonemapping.material.bin"
+        ),
+        [Parameter(Mandatory = $false)]
+        [string]$BackupDir = "$BRTX_DIR\backup"
+    )
+
+    Backup-InitialShaderFiles -Location $Location -Materials $Materials -BackupDir $BackupDir
+
+    # Get base name of location
+    $instance = ($Location -split "\\")[-1].Replace(" ", "_")
+
+    # Convert backup dir to zip archive
+    $zipFilename = "betterrtx_backup_" + ($instance) + (Get-Date -Format "yyyy-MM-dd_HH-mm") + ".zip"
+
+    # Prompt user for output directory
+    $dialog = New-Object System.Windows.Forms.FolderBrowserDialog
+    $dialog.Description = $T.backup_instance_location + " `"$instance`""
+
+    if ($dialog.ShowDialog() -ne [System.Windows.Forms.DialogResult]::OK) {
+        return $false
+    }
+
+    $zip = Join-Path -Path $dialog.SelectedPath -ChildPath $zipFilename
+
+    if (Test-Path $zip) {
+        Remove-Item -Path $zip -Force
+    }
+
+    Compress-Archive -Path $BackupDir -DestinationPath $zip -Force
+    
+    Remove-Item -Path $BackupDir -Force -Recurse
+
+    # Rename it to .mcpack so it can be used with the installer again
+    Rename-Item -Path $zip -NewName ($zip -replace ".zip", ".mcpack") -Force
+
+    return $true
+}
+
+function Copy-ShaderFiles() {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$Location,
+        [Parameter(Mandatory = $true)]
+        [string[]]$Materials
+    )
+
+    $mcDest = "$Location\data\renderer\materials"
+
+    $isSideloaded = $Location -notlike "C:\Program Files\WindowsApps\"
```

Comment:
This line might be throwing an error because "C:\Program Files\WindowsApps\" should be "C:\Program Files\WindowsApps\*"

Reply:
It looks like these errors are due to running an older version of the script, plus there is likely encryption on that path.

There may be a similar error once you run the latest version. I don't know what to expect if it is unable to copy due to encryption